### PR TITLE
Hyperzine increases blood loss

### PR DESCRIPTION
**Pull request by ghost**
have nothing against hyperzine but I think it's a cool side effect

it's an anti-coagulant or something idk

:cl:
 * tweak: Hyperzine increases internal and external bleeding by roughly 30%

### CODE REVIEW

**Comment by ShiftyRail** [comment]
Candle that burns twice as bright, lasts half as long...

**Comment by Manleto** [comment]
Should add something to affect Diona, like an even higher venerability to burn

**Comment by beesil** [comment]
Do you have any data on how much this actually affects blood loss? I think the percentage could be increased, but overall I'm glad for hyperzine to have a downside.

**Comment by PrimeDSS13** [comment]
good PR
I'd buff it to 20% honestly, 10% is pretty negligible.

**Comment by cloutcode** [comment]
does it happen with coke too

**Comment by ghost** [comment]
> 
> 
> Do you have any data on how much this actually affects blood loss? I think the percentage could be increased, but overall I'm glad for hyperzine to have a downside.

I didn't actually run tests on it but I believe the values I changed should result in a directly proportional % increase. Bleeding code could probably do with a refactor to be clearer, but that's a task for another day.


> 
> 
> good PR
> I'd buff it to 20% honestly, 10% is pretty negligible.

I originally set it to 20% but lowered it since I wasn't sure how people would react to more substantial change. I might change it back since 20% isn't too big of an increase. Inapprovaline DECREASES blood loss by 30% and biofoam completely stops it, so it probably wouldn't be that big of a deal anyways.


> 
> 
> Should add something to affect Diona, like an even higher venerability to burn

I don't think there's an easy way to do this, but I'll think on it and see what I can do. They definitely shouldn't get off even easier when it benefits them the most.


> 
> 
> does it happen with coke too

Nope, adding it would be easy but I'd consider it another bonus of using All Natural(tm) chems.

**Comment by DamianX** [comment]
50%

**Comment by Dacendeth** [comment]
Raise it up.

**Comment by ghost** [comment]
It's increased to 30%, but this involved refactoring the code for bleeding a bit since it previously didn't do proper % modifiers. Inaprovaline should also slow down bleeding by 30% for both external and internal now (previously it was only internal, external was commented out). Not counting the hyperzine addition the end result of these changes should be pretty close to before.

**Comment by Anton-Kr** [comment]
Dionas already dont get healed on hyperzine

**Comment by strawberrytan** [comment]
Honestly when I'd heard 10% it sounded like fucking nothing, 30% is closer to a good number but I'd argue even 50% would be interesting to see

On a 10% / 30% / 50% blood loss increase, someone who was going to lose 10%, 20%, 30% blood would be seeing instead a loss of (11/12/15)%, (22/24/30)%, (33/39/45)% blood loss respectively

TL;dr I really like the idea since raising to 30% since 10% is fucking nothing but it would be interesting to even see 50% tried out

**Comment by drawsstuff** [comment]
This isn't something that stimulants do but it's too cool to pass up. Hyperzine being used to treat random event blood-clots when?

**Comment by Eneocho** [comment]
Does blood loss correlate to the heart's BPM codewise? Instead of being an anticoagulant, it could raise the heartbeat, speeding up the blood loss. Then again, you could just make hyperzine-inaprovaline pills to slow/stop the bleeding.

**Comment by ghost** [comment]
> Does blood loss correlate to the heart's BPM codewise? Instead of being an anticoagulant, it could raise the heartbeat, speeding up the blood loss. Then again, you could just make hyperzine-inaprovaline pills to slow/stop the bleeding.

No, blood loss occurs every Life() tick (which is every two seconds iirc) based on the severity of wounds.

**Comment by Eneocho** [comment]
> 
> 
> > Does blood loss correlate to the heart's BPM codewise? Instead of being an anticoagulant, it could raise the heartbeat, speeding up the blood loss. Then again, you could just make hyperzine-inaprovaline pills to slow/stop the bleeding.
> 
> No, blood loss occurs every Life() tick (which is every two seconds iirc) based on the severity of wounds.

Damn. It should correlate to both severity and BPM. That way, a bleeder would be more or less severe depending on whether the guy was on stimulants or depressants. Feed the guy an SNRI (Tramadol, for example) and his heart rate slows down, making the bleeder less severe. Feed him a stimulant (Cocaine or Hyperzine) and the bleeder starts gushing.

**Comment by mrflarg** [comment]
Blessed PR, who knew ancient would be the one to stand against the hyperzine fiends, il leave flowers on your grave when you inevitably get your skull smashed in from an angry hyperzine addict

**Comment by Eneocho** [comment]
On my way to make hyperzine-inaprovaline pills and keep the fiends going.
